packages/autocomplete-core/src/resolve.ts

```diff
@@ -171,7 +171,7 @@ export function postResolve<TItem extends BaseItem>(
       : results;
 
```

Comment:
To avoid getting the confusing `[null]` invariant, perhaps we can have another invariant right above that one, for the special case of `getItems` returning `undefined`:

```suggestion
    invariant(
      (items as Array<typeof items>).every(Boolean),
      `The \`getItems\` function must return an array of items but returned ${JSON.stringify(undefined)}.
      
      Did you forget to return items?
      
      See: https://www.algolia.com/doc/ui-libraries/autocomplete/core-concepts/sources/#param-getitems
    );
```

Reply:
Yes good call. I've added it after the first invariant, because this one assumes for `items` to be an array (of possibly `undefined` values). The first invariant throws if `items` isn't an array, so as we reach the second one we're covered.

See bdc5b95b.